src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -1725,7 +1725,7 @@ public virtual bool Accepts(DType type, out KeyValuePair<string, DType> schemaDi
                     break;
 
                 case DKind.Enum:
-                    accepts = (Kind != type.Kind && type.Kind == DKind.Unknown) ||
+                    accepts = (Kind != type.Kind && (type.Kind == DKind.Unknown || EnumSuperkind == type.Kind)) ||
```

Comment:
The line below already covers the case where we want two enums to be compatible if the EnumSuperKind is the same for both. If the line below is failing, should we instead set the EnumSuperKind of an enum type to itself?

Reply:
In the example listed in the PR description (`Switch(TextInput1.DisplayMode, DisplayMode.Edit, "editing", DisplayMode.View, "viewing", "other")`), when we use `DisplayMode.Edit` or `DisplayMode.View` we don't get an enum value; instead, we get the actual (string, in this case) value of the enum. So the comparison is "does DisplayMode enum accept a string"? Since "string" doesn't have an EnumSuperKind, the line below will return false.

Reply:
Then why does your change fix the problem?

Reply:
In that example (line 1728), [this.]Kind == DKind.Enum, type.Kind == DKind.String, and [this.]EnumSuperkind == DKind.String.